### PR TITLE
build: Update ffmpeg config for CUDA

### PR DESCRIPTION
**Pull request by j0sh**
**What does this pull request do? Explain your changes. (required)**
Splits out the ffmpeg-related updates from https://github.com/livepeer/go-livepeer/pull/1124 in order to make that PR less unwieldy.

**Specific updates (required)**

- Updates LPMS (and associated joy4) . Requires https://github.com/livepeer/lpms/pull/160 and https://github.com/livepeer/joy4/pull/6
- Update ffmpeg to use Livepeer forked version
- Add h264_cuvid to ffmpeg configure.
- Use LLVM instead of cuda nvcc
- Update unit tests that break as a result of the ffmpeg update
- Misc configure flag cleanup

**How did you test each of these updates (required)**

Existing unit tests

**Does this pull request close any open issues?**

**Checklist:**

- [ ] README and other documentation updated
- [ ] Node runs in OSX and devenv
- [x] All tests in `./test.sh` pass

### CODE REVIEW

**Review by darkdarkdragon — COMMENTED** [review]
Looks good, but gave `ERROR: cuda_llvm requested but not found` in CI

File: install_ffmpeg.sh [hunk 1]
```diff
@@ -103,7 +103,7 @@ EXTRA_LDFLAGS=""
 if [ $(uname) == "Linux" ]; then
   if [ -e /usr/local/cuda/include ]; then
```
Comment (iameli) [inline install_ffmpeg.sh]: This isn't a good check anymore, as `clang` doesn't require CUDA as a build-time dependency. I went with

```
if which clang > /dev/null; then
```

Really what we want to check for is clang >= 8.0, but I dunno how to do that in shell.

**Comment by j0sh** [comment]
> ERROR: cuda_llvm requested but not found in CI
> This isn't a good check anymore

Both concerns should be addressed with the merge here of https://github.com/livepeer/go-livepeer/pull/1207 - let's see if CI passes.

**Comment by iameli** [comment]
It did! 🎉

**Review by iameli — APPROVED** [review]

**Comment by j0sh** [comment]
Rebased and merged, thanks!

**Pull request merged**